packages/ui/src/models/camel/__snapshots__/kamelet-resource.test.ts.snap

```diff
@@ -37,6 +37,7 @@ exports[`KameletResource should convert to JSON 1`] = `
       "camel:kamelet",
     ],
     "template": {
+      "beans": undefined,
```

Comment:
I'm not 100% sure what's going on, but the snapshot test got to fail because of this difference. And agree I believe there's no `undefined` in real JSON/YAML so it has to be removed when it's persisted.